aio_pika/queue.py

```diff
@@ -170,6 +174,28 @@ def consumer(channel: Channel, envelope, properties, body: bytes):
 
     @BaseChannel._ensure_channel_is_open
     @asyncio.coroutine
+    def cancel(self, consumer_tag: ConsumerTag, timeout=None):
+        """ This method cancels a consumer. This does not affect already
+        delivered messages, but it does mean the server will not send any more
+        messages for that consumer. The client may receive an arbitrary number
+        of messages in between sending the cancel method and receiving the
+        cancel-ok reply. It may also be sent from the server to the client in
+        the event of the consumer being unexpectedly cancelled (i.e. cancelled
+        for any reason other than the server receiving the corresponding
+        basic.cancel from the client). This allows clients to be notified of
+        the loss of consumers due to events such as queue deletion.
+
+        :param consumer_tag: consumer tag returned by :func:`~aio_pika.Queue.consume`
+        :param timeout: execution timeout
+        :return: Basic.CancelOk when operation completed successfully
+        """
+        f = self._create_future(timeout)
+        self._channel.basic_cancel(f.set_result, consumer_tag=consumer_tag, nowait=False)
```

Comment:
I think you set `nowait` to `false` to prevent the future from never be callbacked by pika? Don't you think this option should be settable by the user? (e.g. `if nowait: f.set_result(None)`? That might be a bit confusing though.)